daliuge-engine/dlg/droputils.py

```diff
@@ -283,6 +319,26 @@ def load_numpy(drop, allow_pickle=False) -> np.ndarray:
     dropio.close()
     return res
 
+
+# def save_jsonp(drop: PathBasedDrop, data: Dict[str, object]):
+#     with open(drop.path, 'r') as f:
+#         json.dump(data, f)
+
+
+# def save_json(drop: DataDROP, data: Dict[str, object]):
+#     # TODO: support BinaryIO or TextIO interface from DataIO?
+#     json_string = json.dumps(data)
+#     drop.write(json_string.encode('UTF-8'))
+
+
+# def load_json(drop: DataDROP) -> dict:
+#     dropio = drop.getIO()
+#     dropio.open(OpenMode.OPEN_READ)
+#     data = json.loads(dropio.buffer())
+#     dropio.close()
+#     return data
+
+
```

Comment:
I've commented this as it reduces the codecoverage however I've been testing this in plugin components. If I have time I'll restore and unit test each of these.